### PR TITLE
feat(instructor): instrumentation

### PR DESCRIPTION
**Pull request by harrisonchu**
adds instrumentation for the instructor library.

TODOs in the PR still:

- [x] sync
- [x] async
- [x] streaming
- [x] context attribute and config support

### CODE REVIEW

File: python/instrumentation/openinference-instrumentation-instructor/README.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+# OpenInference crewAI Instrumentation
```
Comment (anticorrelator) [inline python/instrumentation/openinference-instrumentation-instructor/README.md:1]: this is the wrong example I think

**Comment by harrisonchu** [comment]
Screenshot 2024-08-02 at 2 42 59 PM

work in progress so far.  open questions note to self: 

- is "TOOL" the right span kind?
- need to make sure async / streaming is covered

File: python/instrumentation/openinference-instrumentation-instructor/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# OpenInference Instructor Instrumentation
+
+[![pypi](https://badge.fury.io/py/openinference-instrumentation-crewai.svg)](https://pypi.org/project/openinference-instrumentation-instructor/)
```
Comment (21ShisodeParth) [inline python/instrumentation/openinference-instrumentation-instructor/README.md:3]:
```suggestion
[![pypi](https://badge.fury.io/py/openinference-instrumentation-instructor.svg)](https://pypi.org/project/openinference-instrumentation-instructor/)
```

Comment (harrisonchu) [reply python/instrumentation/openinference-instrumentation-instructor/README.md:3]: thanks man, nice catch :)

**Review by axiomofjoy — APPROVED** [review]

File: python/instrumentation/openinference-instrumentation-instructor/src/openinference/instrumentation/instructor/_wrappers.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+            span_name = f"{instance.__class__.__name__}.{wrapped.__name__}"
+        else:
+            span_name = wrapped.__name__
+        with self._tracer.start_as_current_span(
+            span_name,
+            attributes=dict(
+                _flatten(
+                    {
+                        OPENINFERENCE_SPAN_KIND: OpenInferenceSpanKindValues.TOOL,
+                        INPUT_VALUE_MIME_TYPE: "application/json",
```
Comment (mikeldking) [inline python/instrumentation/openinference-instrumentation-instructor/src/openinference/instrumentation/instructor/_wrappers.py:200]: There's also tool definition attributes that probably should be captured

File: python/instrumentation/openinference-instrumentation-instructor/src/openinference/instrumentation/instructor/version.py [hunk 4]
```diff
@@ -0,0 +1 @@
+__version__ = "0.1.0"
```
Comment (mikeldking) [inline python/instrumentation/openinference-instrumentation-instructor/src/openinference/instrumentation/instructor/version.py:1]:
```suggestion
__version__ = "0.0.1"
```

**Review by mikeldking — APPROVED** [review]

File: python/instrumentation/openinference-instrumentation-instructor/src/openinference/instrumentation/instructor/_wrappers.py (hunk 3 above)
Comment (harrisonchu) [reply python/instrumentation/openinference-instrumentation-instructor/src/openinference/instrumentation/instructor/_wrappers.py:200]: ack -- will get this done in follow up

**Pull request merged**